### ISSUE TITLE
Reduce noise for screen reader users 

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#172465]
When a screen reader user creates or focuses a terminal, they hear the `aria-label` that we set which includes keybindings for how to navigate in the terminal etc. 

Since creating terminals is a pretty frequent occurrence, I imagine that they will have heard this info repeatedly within their first few uses of VS Code. 

Should we have a way for screen reader users to disable `aria-label` meta info to reduce the noise they must filter through to hear relevant/important info?

Should we just remove the extra info after creating X terminals?

The editor should follow suit here - ATM, the same info is presented repeatedly. 

@jooyounseo @isidorn would love to hear your thoughts and feedback - would this be of interest to screen reader users?

**Comment by isidorn** [comment]
@meganrogge cool idea. I think I was debating something similar - that the aria label of the editor should not be read all the time. But so far no user has complained about this. So I never pursued this more. Overall I think this makes sense, but in practice this might be useful, and when not users just press `esc` to stop the announcment.

**Comment by meganrogge** [comment]
@jooyoungseo thoughts?

**Comment by meganrogge** [comment]
Let's include this as a question in our survey #171199

**Comment by jooyoungseo** [comment]
I am on the fence because some novice users might find the instructions provided by aria-label helpful whereas some experienced uses would find it redundant. That's why some popular screen reader, such as JAWS, have verbosity setting where users can customize the control instruction details following their needs. However, aria-label cannot easily be filtered through screen reader itself. Maybe, VSCode team can implement an a11y verbosity setting to toggle the aria-label attachment.

**Comment by jvesouza** [comment]
It would be great, at least in my opinion, to have a way to inhibit this information after some time of use. Perhaps an option in accessibility settings.

**Comment by meganrogge** [comment]
@Tyriar suggested we could possibly pull the setting from the OS and use that within VS Code instead of adding one

**Comment by meganrogge** [comment]
@jooyoungseo suggested that it would be great if audio cues could be played in place of reading "left bracket", "period" and other punctuation

**Comment by jooyoungseo** [comment]
To be clear, replacing punctuations with earcons are possible with screen readers. [mltony/nvda-phonetic-punctuation](https://github.com/mltony/nvda-phonetic-punctuation) is a good example. However, there are some color coding and semantics in VSCode to indicate their symbol groups, such as function, class, variable, etc. which are all ignored in screen reader. It would be so nice if there would be a way to provide such information to screen reader. Voice-fonts (using different speech syntehsizer for different semantic symbols) would be nice, but we may need to work closely with each screen reader vendors to achieve this.

Currently, it is possible to force screen readers to switch between different syntehsizer engine by using lang attribute: https://github.com/mltony/nvda-phonetic-punctuation

Some synthesizers support this, such as [Nuance Vocalizer](https://www.nuance.com/omni-channel-customer-engagement/voice-and-ivr/text-to-speech/vocalizer/language-availability.html).

[Emacspeak](https://www.semanticscholar.org/paper/Emacspeak%E2%80%94a-speech-interface-Raman/faf8817fcd88a80bfedc0569416d899ed58d4b67) is one of the most accessible editors that uses both voice-fonts and earcons.

**Comment by meganrogge** [comment]
With the initial small set that we allow configuring in #180365, we should find other components that would benefit from:

1. a setting like this
2. an `aria-label` with helpful information and then a setting like this

**Comment by lramos15** [comment]
![Recording 2023-04-26 at 09 01 45](https://user-images.githubusercontent.com/4544166/234634511-75a1ac26-2753-4e8b-a0c8-4ee19de168ab.gif)

Seems to work for terminal, but not diff.

**Comment by meganrogge** [comment]
are you sure you had saved the setting? it's working for me @lramos15 

https://user-images.githubusercontent.com/29464607/234640116-50c9f4bb-89f9-4bde-967c-0255fe0ef319.mov

**Comment by lramos15** [comment]
Yes I'm sure the setting is saved. I created a diff by right clicking a file in the explorer and selecting "select for compare" and then selecting a different one and doing "compare with selected"

**Comment by jvesouza** [comment]
At the beginning of the tests I had the feeling that it was not working correctly, but after repeating the tests a few times I came to the conclusion that everything was right.

Would it be possible to clarify the meaning of the message 'undefined use Shift + F7 to navigate changes'? More specifically why the word undefined in the message?

**Comment by meganrogge** [comment]
Yes @jvesouza making a new issue for that and should be easy to fix. Thanks for testing
